src/inabox/inabox-iframe-messaging-client.js

```diff
@@ -51,13 +53,25 @@ function createIframeMessagingClient(win) {
   }
   iframeClient.setSentinel(sentinel || getRandom(win));
 
-  // Bet the top window is the scrollable window and loads host script.
-  // TODO(lannka,#9120):
-  // 1) check window ancestor origin, if the top window is in same origin,
-  // don't bother to use post messages.
-  // 2) broadcast the request
-  iframeClient.setHostWindow(win.top);
-  return iframeClient;
+  iframeClient.setBroadcastMode(true);
+  return new Promise(resolve => {
+    let hostWin = win;
+    let j = 0;
+    const unlisten = iframeClient.registerCallback(MessageType.HOST_RESPONSE,
```

Comment:
I don't see any references to `unlisten`?

Reply:
it's called [here](https://github.com/ampproject/amphtml/pull/21208/files/dca34f779ac80acf1012b69284ab341e5db9660a#diff-ea1167497587e3825e660a8b8293a9feR64).